### PR TITLE
Code formatting configs for tslint and prettier are moving to published packages.

### PR DESCRIPTION
**Pull request by dzearing**
We'd like to have partners be able to share and contribute to our shared tslint and prettier rules to encourage more coding consistency across packages.

We will publish this once it's sharable and ready, but this is the first stab. I'll add some README.md files in a separate pr.

### CODE REVIEW

File: packages/tslint-rules/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "office-ui-fabric-react-tslint",
+  "name": "@uifabric/tslint-rules",
   "version": "6.0.0",
   "description": "Shared tslint rules",
```
Comment (cliffkoh) [inline packages/tslint-rules/package.json]: I feel like leaf dependencies like this which will rarely change should not be tied to Fabric version. They should adhere to semver particularly if they're meant to be used standalone in other repos.

File: packages/prettier-rules/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@uifabric/prettier-rules",
+  "version": "6.0.0",
+  "description": "Shared Prettier rules",
```
Comment (cliffkoh) [inline packages/prettier-rules/package.json]: Same comment - I think this should follow semver. With leaf dependencies, you'll never run into the so-called "diamond dependency" problem.

File: rush.json [hunk 3]
```diff
@@ -141,8 +141,13 @@
       "shouldPublish": false
     },
     {
-      "packageName": "office-ui-fabric-react-tslint",
-      "projectFolder": "packages/office-ui-fabric-react-tslint",
+      "packageName": "@uifabric/tslint-rules",
+      "projectFolder": "packages/tslint-rules",
+      "shouldPublish": false
+    },
```
Comment (cliffkoh) [inline rush.json]: I'm surprised at shouldPublish: false. Is this something you plan to flip after manual publish?

File: common/changes/@uifabric/charting/code-formatting_2018-09-24-22-01.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/charting",
+      "comment": "Moving tslint/prettier dependencies ",
+      "type": "patch"
+    }
```
Comment (cliffkoh) [inline common/changes/@uifabric/charting/code-formatting_2018-09-24-22-01.json]: All changes to all packages are dev-dependencies only and have 0 impact on consuming apps. Consequently these should technically all be of type none. I recommend doing a find and replace.

**Comment by cliffkoh** [comment]
You have an extra z in your commit message.
image

~~Literally unplayable.~~

File: packages/tslint-rules/package.json (hunk 1 above)
Comment (dzearing) [reply packages/tslint-rules/package.json]: I've moved it to 1.0.0

File: packages/prettier-rules/package.json (hunk 2 above)
Comment (dzearing) [reply packages/prettier-rules/package.json]: moved to 1.0.0

**Comment by cliffkoh** [comment]
@dzearing the changefile feedback was not addressed consequently resulting in a lot of redundant publishing for all the packages :( 

It would have been a really trivial thing to follow as well with just a find and replace.

**Pull request merged**